langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/Map.java

```diff
@@ -60,7 +61,13 @@ public static BXml map(BXml x, BFunctionPointer<Object, Object> func) {
         AtomicInteger index = new AtomicInteger(-1);
         AsyncUtils.invokeFunctionPointerAsyncIteratively(func, null, METADATA, x.size(),
                 () -> new Object[]{x.getItem(index.incrementAndGet()), true},
-                result -> elements.add((BXml) result), () -> ValueCreator.createXmlSequence(elements),
+                result -> {
+                    if (result instanceof XmlSequence xmlSequence) {
```

Comment:
Instead of using `instanceof`, use the `getNodeType` function and check if the node type equals `sequence`.